### ISSUE TITLE
[Bug] Can't save new, empty text file

### ISSUE THREAD

**Issue by RonnyAL** [JasonStein/Notepads#291]
I realize that this isn't a typical use case, but sometimes with "regular" Notepad I've saved empty text files before typing anything at all because I already know where I want to keep the file and what I want to call it although it doesn't contain any text yet.

In Notepads, saving is disabled when the file has no content - except if you write some text, save it and then remove the text again. In that case, you're able to save the empty file. In my opinion, saving should be enabled at all times, regardless of the contents of the file.

**Comment by JasonStein** [comment]
You can use "save as" when there is no text in the new document. This is the recommended way.

**Comment by RonnyAL** [comment]
@JasonStein Thanks for commenting on this so soon.

It's good to know that there is an alternative. For me personally it feels wrong since I almost exclusively use keyboard shortcuts. So Ctrl+Shift+S works with an empty text document, but in my mind Ctrl+Shift+S is meant for saving something that's already saved in a new file. So I'd have to mentally rearrange things for this particular software.

Even semantically speaking, it makes more sense to me to simply "save" a newly created document rather than to "save [it] as" something when it is still unsaved. Lastly, this is pretty much standardized behaviour for any kind of (text/image/audio/video) editing software.

With that in mind, consider this a suggestion. I understand if it's not a priority for now but would appreciate if you consider taking a closer look. Otherwise - if I find the time - I might create a pull request for it in the future.

In any case, thanks for creating and sharing this awesome piece of software! :) Been waiting for something like this to come along.

**Comment by JasonStein** [comment]
Thanks for the advice. If you use Windows Notepad, you will notice that when you hit "ctrl+s" for empty document, the popup window actually is calling "Save As":
![image](https://user-images.githubusercontent.com/1166162/74146829-d3564980-4bb6-11ea-90aa-e9230fe0160a.png)
As well for notepad++, the "Save" button is also disabled when document is empty:
![image](https://user-images.githubusercontent.com/1166162/74146902-f5e86280-4bb6-11ea-9016-bcb536bd3261.png)
I think this is well established "behavior", so I just implement as the same.

**Comment by JasonStein** [comment]
Also, Visual Studio Code:
![image](https://user-images.githubusercontent.com/1166162/74147022-39db6780-4bb7-11ea-8712-5ce1844c0861.png)

**Comment by RonnyAL** [comment]
You're right! Actually learned something new today, then. I guess _Save as_ makes sense since _Save_ typically refers to saving modifications, whereas _Save as_ refers to saving something hitherto unsaved. After some testing I found that both VS Code and Windows Notepad will bring up the _Save as_ dialog if you press Ctrl+S in an unsaved document - regardless of its content (or lack thereof). Only Notepad++ among these behaves like Notepads, where the _Save_ option becomes available only after adding some content (and optionally, removing it again).

So I apparantly misspoke earlier. Sorry about that. The convention in the scenario described isn't to bring up the _Save_ dialog, but the _Save as_ dialog as you stated. Is that something you'd consider? To clarify - I'm suggesting enabling the Ctrl+S shortcut immediately but having it open the _Save as_ dialog if the document hasn't been saved yet.

Finally, in asking you to "consider" this I don't mean to come across as demanding. If that's how it comes across, it would be because I have failed to translate my meaning into English (being foreign). Like I mentioned earlier, I'd be happy to submit a pull request if I figure out how to implement the changes requested. However, I'd still be interested in knowing if you'd be willing to accept such a pull request before even making an attempt.
